Fix for unhandled undefined config

bryandel opened the pull request:
When an invalid application id is passed either for reset/change password or email verification, config.get returns undefined. This causes internal server.

montymxb commented:
Nice catch. The standard 'Not found.' & '404' response seems, @flovilmart think we're good using the existing `missingPublicServerURL` for this as well?

flovilmart commented:
We should probably throw if Config.get(appId) don’t return anything no?

montymxb commented:
Something like this?
```js
throw new Parse.Error(Parse.Error.OTHER_CAUSE, 'appId not found');
```
or maybe `OBJECT_NOT_FOUND`? Not too many codes we could match to this unless we wanted to use `INTERNAL_SERVER_ERROR`.

flovilmart commented:
What do we usually throw when making a request with an invalid appId? Unauthorized I guess .)

montymxb commented:
No specific cases come to mind actually, just `You must provide an appId!` in **ParseServer.js** and `...appIds or appSecret is incorrect.` in **vkontakte.js**. What if we made the message just 
```
appId is incorrect
```
Simple and to the point.

montymxb commented on src/Routers/PublicAPIRouter.js:
```diff
@@ -15,7 +15,7 @@ export class PublicAPIRouter extends PromiseRouter {
     const config = Config.get(appId);
 
-    if (!config.publicServerURL) {
+    if (!config || !config.publicServerURL) {
```
Would you be willing to update this to handle `!config` separately? We would like to throw an exception rather than return a standard 404. Something like this should do
```js
if (!config) {
    throw new Parse.Error(Parse.Error.OTHER_CAUSE, 'appId does not exist in config.');
}
```

flovilmart replied:
following in middleware.js if appId can't be found, this should be 
```
function invalidRequest(req, res) {
  res.status(403);
  res.end('{"error":"unauthorized"}');
}
```

montymxb replied:
Ahh, didn't see that initially. @bryandel take flovilmart's suggestion for the proper error to report.

montymxb commented on src/Routers/PublicAPIRouter.js:
```diff
@@ -40,7 +40,7 @@ export class PublicAPIRouter extends PromiseRouter {
     const config = Config.get(appId);
 
-    if (!config.publicServerURL) {
+    if (!config || !config.publicServerURL) {
```
Same here

montymxb commented on src/Routers/PublicAPIRouter.js:
```diff
@@ -66,7 +66,7 @@ export class PublicAPIRouter extends PromiseRouter {
     return new Promise((resolve, reject) => {
       const config = Config.get(req.query.id);
-      if (!config.publicServerURL) {
+      if (!config || !config.publicServerURL) {
```
and here

montymxb commented on src/Routers/PublicAPIRouter.js:
```diff
@@ -89,7 +89,7 @@ export class PublicAPIRouter extends PromiseRouter {
     const config = req.config;
 
-    if (!config.publicServerURL) {
+    if (!config || !config.publicServerURL) {
```
same

montymxb commented on src/Routers/PublicAPIRouter.js:
```diff
@@ -114,7 +114,7 @@ export class PublicAPIRouter extends PromiseRouter {
     const config = req.config;
 
-    if (!config.publicServerURL) {
+    if (!config || !config.publicServerURL) {
```
same

bryandel commented:
Thank you for the suggestions. I have changed the implementation and also added tests for the code coverage. But it seems in the last build a different test has failed

> update android device token with duplicate device token
>       - Expected 1 to equal 0.

montymxb commented:
@bryandel thanks again for the PR!
